code/controllers/subsystems/zcopy.dm

```diff
@@ -77,10 +81,23 @@ SUBSYSTEM_DEF(zcopy)
 	..("Q:{T:[queued_turfs.len - (qt_idex - 1)]|O:[queued_overlays.len - (qo_idex - 1)]} T:{T:[openspace_turfs]|O:[openspace_overlays]}")
 
 /datum/controller/subsystem/zcopy/Initialize(timeofday)
+	calculate_zstack_limits()
 	// Flush the queue.
 	fire(FALSE, TRUE)
 	return ..()
 
+// If you add a new Zlevel or change Z-connections, call this.
+/datum/controller/subsystem/zcopy/proc/calculate_zstack_limits()
```

Comment:
Not familiar with away site loading et-al, so not sure where this should get called.

Reply:
Seems like they now go through `INCREMENT_WORLD_Z_SIZE`, so at least not hard to find. I'm not super clear on the ordering requirements for this; the connectivity is set in landmark/map_data/New, but if maps are being loaded via dmm_suite and queuing zmimic updates and then sleeping, there's no guarantee that will have been created before things start being queued here, unless you're already handling that. If you are, then can e.g. call this from both that macro and from `map_data/New`.

Reply:
This should only matter for things mutating the world size after Z-Mimic init, if away sites are loaded earlier than that they're probably fine.

Reply:
Well, away sites (including multiz ones) can be loaded via admin verb. There are a couple of other things that add blank z levels.